From 8935be5abe693a9d48a9f40862a83c47ad540521 Mon Sep 17 00:00:00 2001
From: Loc Vu <loc.vu.zn@renesas.com>
Date: Thu, 21 Jan 2021 13:48:52 +0700
Subject: [PATCH 032/135] arm64: dts: renesas: rzg2-smarc: enable SDHI0 for MMC
 function

This commits configures SDHI0 to support MMC function with mode HS200.

Signed-off-by: Loc Vu <loc.vu.zn@renesas.com>
---
 arch/arm64/boot/dts/renesas/rzg2-smarc.dtsi | 38 +++++++++++++++++++++++++++++
 1 file changed, 38 insertions(+)

diff --git a/arch/arm64/boot/dts/renesas/rzg2-smarc.dtsi b/arch/arm64/boot/dts/renesas/rzg2-smarc.dtsi
index 0ac0d9c..f9c872f 100644
--- a/arch/arm64/boot/dts/renesas/rzg2-smarc.dtsi
+++ b/arch/arm64/boot/dts/renesas/rzg2-smarc.dtsi
@@ -18,6 +18,16 @@
 		stdout-path = "serial0:115200n8";
 	};
 
+	reg_1p8v: regulator0 {
+		compatible = "regulator-fixed";
+
+		regulator-name = "fixed-1.8V";
+		regulator-min-microvolt = <1800000>;
+		regulator-max-microvolt = <1800000>;
+		regulator-boot-on;
+		regulator-always-on;
+	};
+
 	reg_3p3v: regulator1 {
 		compatible = "regulator-fixed";
 
@@ -53,6 +63,20 @@
 		function = "scif0";
 	};
 
+	sdhi0_pins: sd0 {
+		sd0_data {
+			pins =  "SD0_DATA0", "SD0_DATA1", "SD0_DATA2",
+				"SD0_DATA3", "SD0_DATA4", "SD0_DATA5",
+				"SD0_DATA6", "SD0_DATA7";
+			power-source  = <1800>;
+		};
+
+		sd0_ctrl {
+			pins = "SD0_CLK", "SD0_CMD", "SD0_RST_N";
+			power-source = <1800>;
+		};
+	};
+
 	sdhi1_pins: sd1 {
 		sd1_data {
 			pins =	"SD1_DATA0", "SD1_DATA1", "SD1_DATA2",
@@ -103,6 +127,20 @@
 	status = "okay";
 };
 
+&sdhi0 {
+	pinctrl-0 = <&sdhi0_pins>;
+	pinctrl-1 = <&sdhi0_pins>;
+	pinctrl-names = "default", "state_uhs";
+
+	vmmc-supply = <&reg_3p3v>;
+	vqmmc-supply = <&reg_1p8v>;
+	bus-width = <8>;
+	mmc-hs200-1_8v;
+	non-removable;
+	fixed-emmc-driver-type = <1>;
+	status = "okay";
+};
+
 &sdhi1 {
 	pinctrl-0 = <&sdhi1_pins>;
 	pinctrl-1 = <&sdhi1_pins_uhs>;
-- 
2.7.4

